src/main/scala/com.mediative.eigenflow/process/ProcessSupervisor.scala

```diff
@@ -0,0 +1,47 @@
+package com.mediative.eigenflow.process
+
+import java.util.Date
+
+import akka.actor.SupervisorStrategy.Stop
+import akka.actor.{ Actor, ActorRef, OneForOneStrategy, Props, SupervisorStrategy, Terminated }
+import com.mediative.eigenflow.StagedProcess
+import com.mediative.eigenflow.publisher.MessagingSystem
+
+import scala.util.{ Failure, Success, Try }
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object ProcessSupervisor {
+  def props(process: StagedProcess, date: Option[Date])(implicit messagingSystem: MessagingSystem) = Props(new ProcessSupervisor(process, date))
+
+  case object Done
+  case object Failed
+}
+
+class ProcessSupervisor(val process: StagedProcess, val startDate: Option[Date])(implicit val messagingSystem: MessagingSystem) extends Actor {
+  private val processManager: ActorRef = context.actorOf(Props(new ProcessManager(process, startDate)))
+
+  context.watch(processManager)
+
+  override def supervisorStrategy: SupervisorStrategy = OneForOneStrategy() {
+    case _ => Stop
+  }
+
+  override def receive: Receive = {
+    case ProcessSupervisor.Done => stopSystem(0)
+    case ProcessSupervisor.Failed => stopSystem(1)
+    case Terminated(`processManager`) => stopSystem(1)
+    case message => processManager.forward(message)
+  }
+
+  /**
+   * System shutdown logic.
+   */
+  private def stopSystem(code: Int): Unit = {
+    Try(messagingSystem.stop())
+
+    context.system.terminate().onComplete {
```

Comment:
@dmitri-carpov @WadeWaldron sorry, that was a half-assed comment on my part. I meant actually the 'reaper' pattern which includes using poison pills to safely shut down individual actors: http://letitcrash.com/post/30165507578/shutdown-patterns-in-akka-2 (see section titled 'The Reaper').